src/GraphNode/GraphNodeFactory.php

```diff
@@ -300,8 +300,11 @@ public function castAsGraphNodeOrGraphEdge(array $data, $subclassName = null, $p
             if (static::isCastableAsGraphEdge($data['data'])) {
                 return $this->safelyMakeGraphEdge($data, $subclassName, $parentKey, $parentNodeId);
             }
+
             // Sometimes Graph is a weirdo and returns a GraphNode under the "data" key
-            $data = $data['data'];
+            if (count($data) === 1) {
```

Comment:
maybe add also a check on `&& isset($data['data'])` in case the only present field is not a `data` field